### ISSUE TITLE
Cors issue

### ISSUE THREAD

**Issue by justinmchase** [jamhall/s3rver#346]
I'm not quite sure how to configure cors so I can upload files from a browser correctly.

I'm getting this error:
```
Access to fetch at 'http://localhost:4569/example/6jDdcYvGgWJZgDezh' from origin 'http://localhost:8888' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

I'm basically doing a `PUT http://localhost:4569/example/6jDdcYvGgWJZgDezh` from the browser. Here is the `cors.xml` file I have configured on the server:

```
<CORSConfiguration>
  <CORSRule>
    <AllowedOrigin>*</AllowedOrigin>
    <AllowedMethod>GET</AllowedMethod>
    <AllowedMethod>PUT</AllowedMethod>
    <AllowedMethod>POST</AllowedMethod>
    <AllowedMethod>HEAD</AllowedMethod>
    <AllowedMethod>OPTIONS</AllowedMethod>
    <MaxAgeSeconds>3000</MaxAgeSeconds>
    <AllowedHeader>Authorization</AllowedHeader>
  </CORSRule>
</CORSConfiguration>
```

This is purely for local development so, to be honest, if there was a way to just completely enable it for every domain / method, that would be swell.

**Comment by justinmchase** [comment]
It turns out that my issue was that I was lacking `<AllowedHeader>Content-Type</AllowedHeader>`...

The linked PR does fix the issue where you're not specifying allowed headers as might happen if querying from curl but all I had to do to fix my issue was add that allowed header and it worked. Would be nice to have wildcards for these fields like:

```xml
<CORSConfiguration>
  <CORSRule>
    <AllowedOrigin>*</AllowedOrigin>
    <AllowedMethod>*</AllowedMethod>
    <AllowedHeader>*</AllowedHeader>
    <MaxAgeSeconds>3000</MaxAgeSeconds>
  </CORSRule>
</CORSConfiguration>
```
